java/client/src/main/java/io/vitess/client/cursor/FieldMap.java

```diff
@@ -37,20 +37,36 @@
  */
 public class FieldMap {
   private final List<Field> fields;
-  private final Map<String, Integer> indexMap;
+  private final Map<String, Integer> labelMap;
+  private final Map<String, Integer> nameMap;
+  private final Map<String, Integer> fullNameMap;
 
   public FieldMap(Iterable<Field> fields) {
     this.fields = ImmutableList.copyOf(checkNotNull(fields));
 
-    indexMap = new CaseInsensitiveMap<String, Integer>();
+    labelMap = new CaseInsensitiveMap<String, Integer>();
+    nameMap = new CaseInsensitiveMap<String, Integer>();
+    fullNameMap = new CaseInsensitiveMap<String, Integer>();
     // columnIndex is 1-based.
     int columnIndex = 1;
     for (Field field : this.fields) {
-      String columnLabel = field.getName();
       // If multiple columns have the same name,
       // prefer the earlier one as JDBC ResultSet does.
-      if (!indexMap.containsKey(columnLabel)) {
-        indexMap.put(columnLabel, columnIndex);
+      String columnLabel = field.getName();
+      if (!labelMap.containsKey(columnLabel)) {
+        labelMap.put(columnLabel, columnIndex);
+      }
+      String origName = field.getOrgName();
```

Comment:
the Field may not have these extra values. This is the case when IncludedFields != ALL. So I would check whether the value is null/empty before doing the extra map work and such. Same goes for both new maps.